packages/host/tests/helpers/index.gts

```diff
@@ -736,3 +736,80 @@ export function setupRealmServerEndpoints(
     lookupNetworkService().mount(handleRealmServerRequest, { prepend: true });
   });
 }
+
+export async function assertMessages(
+  assert: Assert,
+  messages: {
+    from: string;
+    message?: string;
+    cards?: { id: string; title?: string; realmIconUrl?: string }[];
+    files?: { name: string; sourceUrl: string }[];
+  }[],
+) {
+  assert.dom('[data-test-message-idx]').exists({ count: messages.length });
+  for (let [index, { from, message, cards, files }] of messages.entries()) {
+    assert
+      .dom(
+        `[data-test-message-idx="${index}"][data-test-boxel-message-from="${from}"]`,
+      )
+      .exists({ count: 1 });
+    if (message != null) {
+      assert
+        .dom(`[data-test-message-idx="${index}"] .content`)
+        .containsText(message);
+    }
+    if (cards?.length) {
+      assert
+        .dom(`[data-test-message-idx="${index}"] [data-test-message-items]`)
+        .exists({ count: 1 });
+      assert
+        .dom(`[data-test-message-idx="${index}"] [data-test-attached-card]`)
+        .exists({ count: cards.length });
+      cards.map(async (card) => {
+        if (card.title) {
+          if (message != null && card.title.includes(message)) {
+            throw new Error(
+              `This is not a good test since the message '${message}' overlaps with the asserted card text '${card.title}'`,
+            );
+          }
+          // note: attached cards are in atom format (which display the title by default)
```

Comment:
Not sure this comment is true. in CardPill there is no delegated render, it just renders its title directly.

Reply:
Correct, I removed the comment.